modules/dop/endpoints/pipeline.go

```diff
@@ -394,8 +409,17 @@ func (e *Endpoints) pipelineCancel(ctx context.Context, r *http.Request, vars ma
 			strutil.Concat(pathPipelineID, ": ", pipelineIDStr)).ToResp(), nil
 	}
 
-	// 根据 pipelineID 获取 pipeline 信息
-	p, err := e.bdl.GetPipeline(pipelineID)
+	// action will cancel pipelineID,  pipelineID not the id that needs to be canceled
+	var cancelRequest apistructs.PipelineCancelRequest
+	if err := json.NewDecoder(r.Body).Decode(&cancelRequest); err != nil {
+		logrus.Errorf("error to decode runRequest")
+	}
+	if r.Header.Get(httputil.InternalActionHeader) == "" {
```

Comment:
Do not depend on internal action header. Other invokers send request body with `CancelRequest` also should cancel pipeline in body.

Reply:
Action why not directly invoke with pipelineID in URL path ?

Reply:
![image](https://user-images.githubusercontent.com/28723047/128129232-cc0440d7-365a-49e5-9b46-16b97f5871c7.png)

the pic is erda action request token, the token will check with pipelineID, if send pipelineID was not this id , request will responce 403 error, i add pipelineID in body when action client request

Reply:
For this reason, the request needs to identify whether it is from the action request. If it is not from the action request, it should follow the original logic

Reply:
You are right! you should comment it in the code.